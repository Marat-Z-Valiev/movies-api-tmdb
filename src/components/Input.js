import React, {useState} from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const InputStyles = styled.input`
	border: 2px solid black;
	padding: 10px;
`;

const Input = ({handleChange, handleClick}) => {
	let [placeholderText, setPlaceholderText] = useState("Enter search query");

	const handleFocus = () => {
		setPlaceholderText((placeholderText = ""));
	};

	const handleOnBlur = () => {
		setPlaceholderText((placeholderText = "Enter search query"));
	};

	const handleKeyPress = (event, handleFunction) => {
		if (event.key === "Enter") {
			handleFunction();
			setPlaceholderText((placeholderText = "Enter search query"));
		}
		return;
	};
	return (
		<InputStyles
			type="text"
			onChange={handleChange}
			onFocus={handleFocus}
			onBlur={handleOnBlur}
			onKeyPress={event => handleKeyPress(event, handleClick)}
			placeholder={placeholderText}
		/>
	);
};

Input.propTypes = {
	// handleChange: PropTypes.func.isRequired,
	handleClick: PropTypes.func.isRequired
};

export default Input;
