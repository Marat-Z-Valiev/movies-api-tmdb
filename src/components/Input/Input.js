import React, {useState} from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const InputStyles = styled.input`
	border: 2px solid black;
	padding: 10px;
	margin-right: 10px;
	border-radius: 15px;
`;

const Input = ({handleChange, handleClick, closeMenu}) => {
	let [placeholderText, setPlaceholderText] = useState("Search");

	const handleFocus = () => {
		setPlaceholderText((placeholderText = ""));
	};

	const handleOnBlur = () => {
		setPlaceholderText((placeholderText = "Search"));
	};

	const runFunctions = () => {
		closeMenu();
		handleClick();
	};

	const handleKeyPress = (event) => {
		if (event.key === "Enter") {
			runFunctions();
			setPlaceholderText((placeholderText = "Search"));
			document.querySelector(".input").value = "";
		}
		return;
	};
	return (
		<InputStyles
			className="input"
			type="text"
			onChange={handleChange}
			onFocus={handleFocus}
			onBlur={handleOnBlur}
			onKeyPress={(event) => handleKeyPress(event)}
			placeholder={placeholderText}
		/>
	);
};

Input.propTypes = {
	handleChange: PropTypes.func.isRequired,
	handleClick: PropTypes.func.isRequired,
};

export default Input;
