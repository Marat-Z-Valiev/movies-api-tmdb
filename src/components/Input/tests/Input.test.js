import React from "react";
import {shallow, configure, mount} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Input from "../Input";

configure({adapter: new Adapter()});

describe("Input", () => {
	const props = {
		handleChange: () => {},
		handleClick: () => {},
		closeMenu: () => {},
	};
	it("should render correctly", () => {
		const wrapper = shallow(<Input {...props} />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});

	it("should trigger the handleChange function on change", () => {
		const props = {
			handleClick: () => {},
			closeMenu: () => {},
		};
		const handleChange = jest.fn();
		const wrapper = shallow(<Input handleChange={handleChange} {...props} />);
		wrapper.simulate("change");
		expect(handleChange).toHaveBeenCalled();
	});

	it("should trigger the handleClick function on click", () => {
		const props = {
			handleChange: () => {},
		};
		const handleClick = jest.fn();
		const closeMenu = jest.fn();
		const wrapper = mount(
			<Input closeMenu={closeMenu} handleClick={handleClick} {...props} />
		);
		const input = wrapper.find("input").at(0);
		input.simulate("keypress", {key: "Enter"});
		expect(handleClick).toHaveBeenCalled();
		expect(closeMenu).toHaveBeenCalled();
	});

	it("should have default placeholder text", () => {
		const wrapper = shallow(<Input {...props} />);
		expect(wrapper.props().placeholder).toBe("Search");
	});
});
