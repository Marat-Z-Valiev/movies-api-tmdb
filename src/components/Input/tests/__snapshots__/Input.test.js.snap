// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<Input__InputStyles
  className="input"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyPress={[Function]}
  placeholder="Search"
  type="text"
/>
`;
